### DISCUSSION TITLE
New opportunity to read/write Dynamic ESS data direct from Victron Cerbo v3.30?

### DISCUSSION THREAD

**Discussion by RSS101** [sfstar/hass-victron#203] [Ideas]
Hoping that I haven’t missed the obvious:

With the full release of the “Dynamic ESS” option with the Cerbo GX update to v3.30, there is a new option to turn on / off the Dynamic ESS setting that is now visible in the Victron core menu settings. Is it possible to add this as an additional sensor/state within the Victron integration to allow this to be controlled and automated via HA?

There is also an option in the new Dynamic ESS menu to select “sell” (i.e. export available energy on demand - solar / battery) or “buy” energy within the automated Dynamic ESS options. Could this data also be added to the Victron integration into HA, to allow these settings to be selected?

I am in the UK and have the necessary permissions in place to export energy to grid for an agreed unit rate.

I’m aware that there is a node-red dynamic ESS solution but this predates the release of v3.30 by Victron. 

Many thanks in advance.

**Comment by sfstar** [comment]
Hello and thank you for opening an discussion item.

Unfortunately victron currently doesn't expose the dynamic ess setting via the modbus server.
In order to get this added you can open a request at the victron repository for the modbus service: https://github.com/victronenergy/dbus_modbustcp
I can add the entity once victron has added this value to the modbus service
